core/common/src/main/java/alluxio/conf/PropertyKey.java

```diff
@@ -1700,18 +1700,27 @@ public String toString() {
           // use jobMasterHostname:jobMasterEmbeddedJournalPort by default.
           .setScope(Scope.ALL)
           .build();
-  public static final PropertyKey MASTER_EMBEDDED_JOURNAL_ELECTION_TIMEOUT =
-      new Builder(Name.MASTER_EMBEDDED_JOURNAL_ELECTION_TIMEOUT)
-          .setDescription(
-              "The election timeout for the embedded journal. When this period elapses without a "
-                  + "master receiving any messages, the master will attempt to become the primary."
-                  + "Election timeout will be waited initially when the cluster is forming. "
-                  + "So larger values for election timeout will cause longer start-up time. "
-                  + "Smaller values might introduce instability to leadership.")
+  public static final PropertyKey MASTER_EMBEDDED_JOURNAL_MIN_ELECTION_TIMEOUT =
+      new Builder(Name.MASTER_EMBEDDED_JOURNAL_MIN_ELECTION_TIMEOUT)
+          .setDescription("The min election timeout for the embedded journal.")
           .setDefaultValue("10s")
           .setConsistencyCheckLevel(ConsistencyCheckLevel.WARN)
           .setScope(Scope.MASTER)
           .build();
+  public static final PropertyKey MASTER_EMBEDDED_JOURNAL_MAX_ELECTION_TIMEOUT =
+      new Builder(Name.MASTER_EMBEDDED_JOURNAL_MAX_ELECTION_TIMEOUT)
+          .setDescription(String.format(
+              "The max election timeout for the embedded journal. When a random period between "
+              + "${%s} and ${%s} elapses without a master receiving any messages, the master "
+              + "will attempt to become the primary Election timeout will be waited initially "
+              + "when the cluster is forming. So larger values for election timeout will cause "
+              + "longer start-up time. Smaller values might introduce instability to leadership.",
+              Name.MASTER_EMBEDDED_JOURNAL_MIN_ELECTION_TIMEOUT,
+              Name.MASTER_EMBEDDED_JOURNAL_MAX_ELECTION_TIMEOUT))
+          .setDefaultValue("20s")
```

Comment:
@apc999 @ZacBlanco Is there a way to read value of `MASTER_EMBEDDED_JOURNAL_MIN_ELECTION_TIMEOUT` dynamically and set it here after multiplying by 2?

Reply:
AFAIK I don't think there's a "performant" way to make default values of properties depend upon one another. I think theoretically you could do the following in the lambda:

```java
() -> {
  AlluxioProperties props = ConfigurationUtils.defaults();
  return props.get(keyName) * 2;
}
```

This unfortunately causes the default properties hashmap on every call to `ConfigurationUtils.default()`. If we did this for more than a few properties it would become _very_ slow to start the JVM because of all the time spent in static initializers. We can probably make some kind of optimization in another PR which doesn't force a copy and returns a read-only version of the properties instead.

Reply:
Do you mind creating a ticket with some possible directions to tackle this issue?

Reply:
done